Userland/Libraries/LibIPC/Decoder.cpp

```diff
@@ -34,8 +34,6 @@ template<>
 ErrorOr<DeprecatedString> decode(Decoder& decoder)
 {
     auto length = TRY(decoder.decode_size());
-    if (length == NumericLimits<u32>::max())
-        return DeprecatedString {};
```

Comment:
You'd also need to remove the Encoder side that sends `NumericLimits<u32>::max()`, else something like `encode(StringView {})` can't be decoded.

(Not sure what you mean by "send a StringView and receive a DeprecatedString (which is an error anyway)", you can totally `encode("foo"sv)` and `decode<DeprecatedString>()` on the other side)

Reply:
That's a very _incorrect_ way of using this, but I can see where you're coming from.
In which case, the existing code is perfectly fine, no need to change it.